controllers/cluster_controller.go

```diff
@@ -385,7 +385,15 @@ func (r *ClusterReconciler) reconcileResources(
 		return ctrl.Result{}, fmt.Errorf("cannot update role labels on pods: %w", err)
 	}
 
-	if err := r.updateClusterRoleLabelsOnPVCs(ctx, resources.instances, resources.pvcs); err != nil {
+	// updated any labels that are coming from the operator
+	if err := r.updateOperatorLabelsOnInstances(ctx, resources.instances); err != nil {
+		return ctrl.Result{}, fmt.Errorf("cannot update instance labels on pods: %w", err)
+	}
+
+	// TODO: refactor how we handle label and annotation reconciliation. We should handle them in a more programmatic way
```

Comment:
we should move to a design where we compare the expected pod labels and annotations with the current living one and reconcile it.